src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1286,16 +1286,21 @@
 (defn update-latest-report!
   "Given a node name, updates the `certnames` table to ensure that it indicates the
    most recent report for the node."
-  [node]
-  {:pre [(string? node)]}
-  (let [latest-report (:id (first (query-to-vec
-                                    ["SELECT id FROM reports
+  [node report-id report-timestamp]
+  {:pre [(string? node) (integer? report-id)]}
+
+  (let [latest-report-timestamp (:timestamp (first (query-to-vec
+                                    ["SELECT latest_report_timestamp
+                                      FROM certnames
                                       WHERE certname = ?
-                                      ORDER BY producer_timestamp DESC
-                                      LIMIT 1" node])))]
-    (jdbc/update! :certnames
-                  {:latest_report_id latest-report}
-                  ["certname = ?" node])))
+                                      " node])))]
+  ;  (if (sql-ts-after? (to-timestamp report-timestamp)
+  ;                   (to-timestamp latest-report-timestamp))
```

Comment:
I think you want something like

```
(sql-ts-after? latest-report-timestamp (to-timestamp report-timestamp))
```

since the stamp out of the database is in the correct form already